### PR TITLE
query: fix SIGSEGV (threading)

### LINKED ISSUE #3842: tensor-query segmentation faults
In GBS environment, try the SSAT test:

```
(tizen-build-env)@kohaku nnstreamer_query]$ gst-launch-1.0 --gst-plugin-path=${PATH_TO_PLUGIN} tensor_query_serversrc num-buffers=3  port=${PORT1} ! other/tensors,num_tensors=1,dimensions=3:300:300:1,types=uint8 ! tensor_query_serversink  port=${PORT2} async=false  videotestsrc num-buffers=1 ! video/x-raw,format=RGB,height=4,width=4 ! filesink location=marker.log 
Setting pipeline to PAUSED ...
Pipeline is live and does not need PREROLL ...
Redistribute latency...
Redistribute latency...
Pipeline is PREROLLED ...
Setting pipeline to PLAYING ...
WARNING: from element /GstPipeline:pipeline0/GstTensorQueryServerSink:tensorqueryserversink0: Pipeline construction is invalid, please add queues.
Additional debug info:
../subprojects/gstreamer/libs/gst/base/gstbasesink.c(1249): gst_base_sink_query_latency (): /GstPipeline:pipeline0/GstTensorQueryServerSink:tensorqueryserversink0:
Not enough buffering available for  the processing deadline of 0:00:00.020000000, add enough queues to buffer  0:00:00.020000000 additional data. Shortening processing latency to 0:00:00.000000000.
WARNING: from element /GstPipeline:pipeline0/GstTensorQueryServerSink:tensorqueryserversink0: Pipeline construction is invalid, please add queues.
Additional debug info:
../subprojects/gstreamer/libs/gst/base/gstbasesink.c(1249): gst_base_sink_query_latency (): /GstPipeline:pipeline0/GstTensorQueryServerSink:tensorqueryserversink0:
Not enough buffering available for  the processing deadline of 0:00:00.020000000, add enough queues to buffer  0:00:00.020000000 additional data. Shortening processing latency to 0:00:00.000000000.
New clock: GstSystemClock
WARNING: from element /GstPipeline:pipeline0/GstTensorQueryServerSink:tensorqueryserversink0: Pipeline construction is invalid, please add queues.
Additional debug info:
../subprojects/gstreamer/libs/gst/base/gstbasesink.c(1249): gst_base_sink_query_latency (): /GstPipeline:pipeline0/GstTensorQueryServerSink:tensorqueryserversink0:
Not enough buffering available for  the processing deadline of 0:00:00.020000000, add enough queues to buffer  0:00:00.020000000 additional data. Shortening processing latency to 0:00:00.000000000.
^Z00:00.0 / 99:99:99.
[1]+  Stopped                 gst-launch-1.0 --gst-plugin-path=${PATH_TO_PLUGIN} tensor_query_serversrc num-buffers=3 port=${PORT1} ! other/tensors,num_tensors=1,dimensions=3:300:300:1,types=uint8 ! tensor_query_serversink port=${PORT2} async=false videotestsrc num-buffers=1 ! video/x-raw,format=RGB,height=4,width=4 ! filesink location=marker.log
(tizen-build-env)@kohaku nnstreamer_query]$ bg
[1]+ gst-launch-1.0 --gst-plugin-path=${PATH_TO_PLUGIN} tensor_query_serversrc num-buffers=3 port=${PORT1} ! other/tensors,num_tensors=1,dimensions=3:300:300:1,types=uint8 ! tensor_query_serversink port=${PORT2} async=false videotestsrc num-buffers=1 ! video/x-raw,format=RGB,height=4,width=4 ! filesink location=marker.log &
0:00:00.0 / 99:99:99.aku nnstreamer_query]$ 0:00:00.0 / 99:99:99.
0:00:00.0 / 99:99:99.aku nnstreamer_query]$ 0:00:00.0 / 99:99:99.
0:00:00.0 / 99:99:99. location= raw1_%1d.log t. ! queue ! tensor_query_client src-port=${PORT1} sink-port=${PORT2} ! multifilesink location=result1_%1d.log0:00:00.0 / 99:99:99. 
Setting pipeline to PAUSED ...
Pipeline is PREROLLING ...
Redistribute latency...
Redistribute latency...
WARNING: from element /GstPipeline:pipeline0/GstTensorQueryServerSink:tensorqueryserversink0: Pipeline construction is invalid, please add queues.
Additional debug info:
../subprojects/gstreamer/libs/gst/base/gstbasesink.c(1249): gst_base_sink_query_latency (): /GstPipeline:pipeline0/GstTensorQueryServerSink:tensorqueryserversink0:
Not enough buffering available for  the processing deadline of 0:00:00.020000000, add enough queues to buffer  0:00:00.020000000 additional data. Shortening processing latency to 0:00:00.000000000.
Pipeline is PREROLLED ...
Setting pipeline to PLAYING ...
Redistribute latency...
New clock: GstSystemClock
Got EOS from element "pipeline0".
Execution ended after 0:00:24.602458256
Setting pipeline to NULL ...
Got EOS from element "pipeline0".
Execution ended after 0:00:00.001480830
Setting pipeline to NULL ...
Caught SIGSEGV
Freeing pipeline ...
(tizen-build-env)@kohaku nnstreamer_query]$ 
[1]+  Segmentation fault      (core dumped) gst-launch-1.0 --gst-plugin-path=${PATH_TO_PLUGIN} tensor_query_serversrc num-buffers=3 port=${PORT1} ! other/tensors,num_tensors=1,dimensions=3:300:300:1,types=uint8 ! tensor_query_serversink port=${PORT2} async=false videotestsrc num-buffers=1 ! video/x-raw,format=RGB,height=4,width=4 ! filesink location=marker.log
(tizen-build-env)@kohaku nnstreamer_query]$ 
(tizen-build-env)@kohaku nnstreamer_query]$ 
(tizen-build-env)@kohaku nnstreamer_query]$ 

```

TODO: run the same test in Ubuntu native shell for easier debugging.

### PR DESCRIPTION
**Pull request by myungjoo**
1. Do not call pthread_join on 0 pid (thread already freed?)
2. Call cancellable before calling pthread_join so that there is no more
 TCP-receiving thread awaiting for a closed connection.

Fixes #3842

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
CC: @gichan-jang @jaeyun-jung Please check if nnstreamer-edge has similar issue or not.

### CODE REVIEW

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 1]
```diff
@@ -917,6 +924,7 @@ accept_socket_async_cb (GObject * source, GAsyncResult * result,
       nns_loge ("Failed to create message handler thread.");
       nnstreamer_query_close (conn);
       g_free (thread_data);
+      conn->msg_thread = 0;
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:927]: line 925. after calling nnstreamer_query_close(), conn is freed. How about moving this into nnstreamer_query_close()?

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 2]
```diff
@@ -530,7 +530,14 @@ nnstreamer_query_server_data_free (query_server_handle server_data)
       if (sdata->conn_queue) {
         while ((conn_remained = g_async_queue_try_pop (sdata->conn_queue))) {
           conn_remained->running = 0;
-          pthread_join (conn_remained->msg_thread, NULL);
+          if (conn_remained->msg_thread) {
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:533]: Suggestion: stopping thread in close function - move this block into nnstreamer_query_close()

Comment (myungjoo) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: Yes, that looks better. updated.

File: gst/nnstreamer/tensor_query/tensor_query_common.c (hunk 1 above)
Comment (myungjoo) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: moved.

**Comment by myungjoo** [comment]
WAIT. This update recreated SIGSEGV :(

**Comment by myungjoo** [comment]
@jaeyun-jung #3823 still has this code active. #3823's build log:


```
[  316s] ==================================================
[  316s] [PASSED] Test Group transform_clamp Passed
[  316s] 
[  316s] [Starting] nnstreamer_query
[  317s] Caught SIGSEGV
[  317s] exec gdb failed: No such file or directory
[  319s] Caught SIGSEGV
[  319s] exec gdb failed: No such file or directory
[  320s] Caught SIGSEGV
[  320s] exec gdb failed: No such file or directory
[  322s] Caught SIGSEGV
[  323s] Caught SIGSEGV
[  323s] exec gdb failed: No such file or directory
[  324s] Caught SIGSEGV
[  324s] exec gdb failed: No such file or directory
[  325s] Caught SIGSEGV
[  325s] exec gdb failed: No such file or directory
[  325s] ==================================================
```

**Comment by jaeyun-jung** [comment]
@myungjoo plz check GC's PR #3782

**Comment by myungjoo** [comment]
> @myungjoo plz check GC's PR #3782

OH..ok.. then, let's move fast with #3782 and #3823

**Pull request closed without merging**